File: Problem_List/14.Longest_Common_Prefix/code.cpp
#include <vector>;
using namespace std;;


class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        if (strs.empty()) return "";

        for(int i = 0; i < strs[0].size(); i++){
            char c = strs[0][i];
            for(int j = 1; j < strs.size(); j++){
                if(strs[j][i] != c || i == strs[j].size()){
                    return strs[0].substr(0,i);
                }
            }
        }
        return strs[0];
    }
};

File: Problem_List/14.Longest_Common_Prefix/code.py
class Solution(object):
    def longestCommonPrefix(self, strs):
        """
        :type strs: List[str]
        :rtype: str
        """
        pref = strs[0]

        for s in strs[1:]:
            while not s.startswith(pref):
                pref = pref[:-1]
                if not pref:
                    return ""

        return pref

        